pkg/controller/migmigration/validation.go

```diff
@@ -290,35 +291,41 @@ func ensureRegistryHealth(c k8sclient.Client, migration *migapi.MigMigration) (i
 			return nEnsured, message, nil
 		}
 
-		registryStatusUnhealthy, podObj := isRegistryPodUnHealthy(registryPods)
-
+		registryStatusUnhealthy, podObj, state := isRegistryPodUnHealthy(registryPods)
 		if !registryStatusUnhealthy {
 			nEnsured++
 		} else {
 			unHealthyPod = podObj
 			unHealthyClusterName = cluster.ObjectMeta.Name
+			reason = state
```

Comment:
I feel we could drop this re-assignment of `state->reason`

Reply:
@djwhatle Initially I was using `state` directly but that value gets replaced with the value returned by `isRegistryPodUnHealthy` for the registry pods of the last `MigCluster` in the loop (i.e. registry pods of destination cluster). That's why I am using an additional variable scooped beyond the `for` loop following the logic that returns `unHealthyPod` and `UnhealthyClusterName`.

Reply:
Not sure I completely follow, but if you found it was necessary then it's fine with me.